citrination_client/client.py

```diff
@@ -238,6 +238,93 @@ def upload_file(self, file_path, data_set_id, root_path=None):
                 message = {"message": "Upload failed.",
                                    "status": r.status_code}
                 return json.dumps(message)
+        else:
+            return json.dumps({
+                    "message": "No file at specified path {0}".format(source_path)
+                })
+
+    def upload_file(self, file_path, dataset_id, root_path=None):
+        """
+        Upload file to Citrination.
+        :param file_path: File path to upload.
+        :param data_set_id: The dataset id to upload the file to.
+        :return: Response object or return code if the file was not uploaded.
+        """
+        return self.upload(dataset_id, file_path)
+
+    def list(self, dataset_id, glob=".", is_dir=False):
+        """
+        List matched filenames in a dataset on Citrination.
+        :param dataset_id: The ID of the dataset to search for files.
+        :param glob: A pattern which will be matched against files in the dataset.
+        :param is_dir: A boolean indicating whether or not the pattern should match against the beginning of paths in the dataset.
+        :return: Response object or return code if the file was not uploaded.
+        """
+        url = self.api_url + '/datasets/' + str(dataset_id) + '/list_filepaths'
+        data = {
+            "list": {
+                "glob": glob,
+                "is_dir": is_dir
+            }
+        }
+        r = requests.post(url, data=json.dumps(data), headers=self.headers)
+        return json.loads(r.content.decode('utf-8'))
+
+    def matched_file_count(self, dataset_id, glob=".", is_dir=False):
+        list_result = self.list(dataset_id, glob, is_dir)
+        if isinstance(list_result["files"], list):
+            return len(list_result["files"])
+        else:
+            return list_result
+
+    def move(self, dataset_id, original_path, target_path, is_dir=False):
+        return self._move_or_copy(dataset_id, "move", original_path, target_path, is_dir)
+
+    def copy(self, dataset_id, original_path, target_path, is_dir=False):
+        return self._move_or_copy(dataset_id, "copy", original_path, target_path, is_dir)
```

Comment:
You might introduce a modicum of safety by using [enums](https://pypi.python.org/pypi/enum34) for the action, unless there's a desire to avoid pulling in additional dependencies.